server/src/main/java/org/elasticsearch/cluster/metadata/DataStream.java

```diff
@@ -589,6 +518,11 @@ public DataStream(StreamInput in) throws IOException {
         );
     }
 
+    static List<Index> readIndices(StreamInput in) throws IOException {
+        in.readString(); // timestamp field, which is always @timestamp
```

Comment:
I was wondering why you chose to always read and ignore the timestamp field instead of using the version of ES to slowly remove the field from here.

Reply:
I didn't want to change serialization logic in this PR. 
I will do this in a follow up pr as part of a larger change that also changes xcontent, but that change is breaking and I wanted to discuss this internally before going ahead with this.